src/dune_rules/odoc.ml

```diff
@@ -440,8 +440,10 @@ let odocs sctx target =
     in
     let obj_dir = Lib_info.obj_dir info in
     Modules.fold_no_vlib modules ~init:[] ~f:(fun m acc ->
+        if List.mem m ~set:(Modules.entry_modules modules) then
```

Comment:
I'm not sure why this isn't working - though I think rather than checking for membership in `Modules.entry_modules` you should fold over the result of that function. In fact, there's a likely looking function later in the file called `entry_modules_by_lib` - try calling this and then folding over the result instead.

Reply:
>  though I think rather than checking for membership in Modules.entry_modules you should fold over the result of that function.

Sorry! Which function exactly, `odocs` or the anonymous, for some other?

Reply:
`entry_modules_by_lib` was what I was thinking of.

Reply:
You can now checkout this - https://github.com/ocaml/dune/pull/3894/commits/eeed25ccfd95ca9b073422ac19e388d02a9b223e

Reply:
This looks great! All working now then?

Reply:
You'll need to rebase before it's mergable (and, of course, we shouldn't merge this before a release of odoc from master, and even then we need to discuss whether backward-compatibility is important)

Reply:
Alright! :)